examples/requirements.txt

```diff
@@ -0,0 +1,16 @@
+# These requirements are shared for all examples
+# Each example subdir has its own `requirements.txt` with extra deps.
```

Comment:
is there a reason the repo-wide requirements aren't in the repo rootdir? I would expect at least the dev requirements (pre-commit, pytest, etc) to not be under examples.

Reply:
Will move the `requirements.txt` to the top level folder